### ISSUE TITLE
add_device_group API call for type='static' fails with 'HTTP Error 500: We couldn't parse your rule' error

### ISSUE THREAD

**Issue by DanielMuller-TN** [librenms/librenms#13211]
### The problem

When attempting to add a device group with a static device membership via the API the API call fails with a 500: We couldn't parse your rule error. It appears that the API endpoint is attempting to validate an empty rules attribute that's only present when calling the API with type='dynamic'.

If you supply a valid rule the API call completes, the rule is discarded and the static device group is created.

### Output of ./validate.php

```txt
[librenms@libnmsappprd1 ~]$ ./validate.php
====================================
Component | Version
--------- | -------
LibreNMS  | 21.8.0-11-g1544eadd3
DB Schema | 2021_25_01_0129_isis_adjacencies_nullable (216)
PHP       | 7.4.6
Python    | 3.6.8
MySQL     | 10.3.28-MariaDB
RRDTool   | 1.7.0
SNMP      | NET-SNMP 5.8
====================================

[OK]    Composer Version: 2.1.6
[OK]    Dependencies up-to-date.
[OK]    Database connection successful
[OK]    Database schema correct
[INFO]  Detected Dispatcher Service
```

### Anything in the logs that might be useful for us?

```txt
The issue appears to be in ./includes/html/api_functions.inc.php in the add_device_group function. The QueryBuilderParser::fromJson method is called in all cases, whether type='static' or type='dynamic'. As the rules attribute isn't required (or relevant) to a static device group the validation fails.
```

**Comment by murrant** [comment]
@DanielMuller-TN can you test the fix in #13222 ?

**Comment by DanielMuller-TN** [comment]
@murrant sorry for the delay - I've applied #13222 (./scripts/github-apply 13222) and it doesn't fix the issue unfortunately.

```
[dmuller@rhmgmtsvrdev3 tests]$ curl -H 'X-Auth-Token: ..SNIP..' -X POST -d '{"name":"New Device Group","type":"static","devices":[1]}' https://..SNIP../api/v0/devicegroups
{
    "status": "error",
    "message": "We couldn't parse your rule"
}
```

The fix in #13222 still tries to evaluate the dynamic group rules which fails and generates the HTTP/500 error.

From includes/html/api_functions.inc.php in #13222:
```
    // Only use the rules if they are able to be parsed by the QueryBuilder
    $query = QueryBuilderParser::fromJson($data['rules'])->toSql();
    if (empty($query)) {
        return api_error(500, "We couldn't parse your rule");
    }
```

I've submitted a pull request #13212 which includes additional conditional logic to skip the rule validation that's generating the API error however it's failing the merge validation step in conflict with #13222 and, being the novice GitHub user that I am I have no idea how to fix it.
